src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -682,6 +694,83 @@ private static void WritePageableLongRunningOperationMethodBody(CodeWriter write
             }
         }
 
+        private void WritePageableLongRunningOperationConvenienceMethod(LowLevelClientMethod clientMethod, ClientFields fields, ProtocolMethodPaging pagingInfo, RestClientMethod nextPageMethod, OperationLongRunning longRunning, bool async)
+        {
+            using (WriteConvenienceMethodDeclaration(writer, clientMethod.ConvenienceMethod!.Signature, async))
+            {
+                // RequestContext context = FromCancellationToken(cancellationToken);
+                var contextVariable = new CodeWriterDeclaration(KnownParameters.RequestContext.Name);
+                WriteCancellationTokenToRequestContext(writer, contextVariable);
+
+                var convenienceMethod = clientMethod.ConvenienceMethod!;
+                var diagnostic = (convenienceMethod.Diagnostic != null ? convenienceMethod.Diagnostic : clientMethod.ProtocolMethodDiagnostic);
+
+                var createEnumerableMethodSignature = new MethodSignature("CreateEnumerable", null, null, None, new CSharpType(typeof(IEnumerable<>), new CSharpType(typeof(Page<>), clientMethod.ConvenienceMethod!.PagingResponseInfo!.ItemType)), null, new[] { ResponseParameter, NextLinkParameter, PageSizeHintParameter }).WithAsync(async);
```

Comment:
Since this new instance statement takes quite a few parameters, could we use positional parameters for each of the parameters here we are using? This will make the ctor invocation more clear, especially we are passing null for some parameters
```suggestion
                var createEnumerableMethodSignature = new MethodSignature(
                Name: "CreateEnumerable", 
                etc: null,
                etc: null,
                etc: None,
                etc: new CSharpType(typeof(IEnumerable<>),
                etc: new CSharpType(typeof(Page<>),
                etc: clientMethod.ConvenienceMethod!.PagingResponseInfo!.ItemType)),
                etc: null,
                Parameters: new[] { ResponseParameter, NextLinkParameter, PageSizeHintParameter }).WithAsync(async);
```

Reply:
And will we use the `ReturnType` in this created method signature? Because this method signature is using "open form" generic types (such as `typeof(Page<>)` instead of `typeof(Page<string>)` with a concrete type).
Where do we actually make the generic type closed?

Reply:
> And will we use the `ReturnType` in this created method signature? Because this method signature is using "open form" generic types (such as `typeof(Page<>)` instead of `typeof(Page<string>)` with a concrete type). Where do we actually make the generic type closed?

@ArcturusZhang It's used. See the generated code: https://github.com/Azure/autorest.csharp/pull/2975/files#diff-ebfa2940fc7dbefca5bdd3ce030063f23127a440fa5b893328e1d5c167a5ea47R283
```csharp
async IAsyncEnumerable<Page<Project>> CreateEnumerableAsync(Response response, string nextLink, int? pageSizeHint, [EnumeratorCancellation] CancellationToken cancellationToken = default)
```
The second parameter of `CSharpType` constructor is the class template argument type. So the return type is actually `typeof(Page<clientMethod.ConvenienceMethod!.PagingResponseInfo!.ItemType>)`